pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java

```diff
@@ -140,6 +140,10 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
             directProxyHandler.outboundChannel.close();
         }
 
+        if (client != null) {
+            client.close();
+        }
```

Comment:
I don't understand why we want to close the entire client at this point. 

When this connection gets closed, we want its paired connection to be closed as well. This should already done at line 140 here above.

Reply:
The client connection Pool is used by lookupProxyHandler and not the directProxyHandler.

Reply:
But the client is shared across all the ProxyConnection instances

Reply:
We changed the logic to have one client per ProxyConnection 
https://github.com/apache/incubator-pulsar/pull/1707

Reply:
Uhm, that doesn't sound a good option to me